backend/SlothOrganizer/SlothOrganizer.Services/Tasks/TaskService.cs

```diff
@@ -40,5 +39,33 @@ public async Task<List<TaskDto>> Get(long dashboardId)
             var tasks = await _taskRepository.Get(dashboardId);
             return _mapper.Map<List<TaskDto>>(tasks);
         }
+
+        public async Task<TaskDto> Update(UpdateTaskDto updateTaskDto)
+        {
+            await _taskCompletionService.Update(updateTaskDto.TaskCompletion);
+            var task = _mapper.Map<UserTask>(updateTaskDto.Task);
+            var updatedTask = _mapper.Map<TaskDto>(await _taskRepository.Update(task));
```

Comment:
As I see both _taskCompletionService.Update and _taskRepository.Update could be called together with WhenAll.

Reply:
It is important that completion is updated before task retrieval, so to run updates simultaneously we would have to add separate call to get updated task